### PR TITLE
Document changeStringEnumsToEnumShapes transformer

### PR DESCRIPTION
**Pull request by srchase**
Adds docs for changeStringEnumsToEnumShapes transformer, including behavior from https://github.com/awslabs/smithy/pull/1739

Additionally, it updates the `changeTypes` documentation to include its ability to transform Strings with `@enum` traits to Enums.

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: docs/source-2.0/guides/building-models/build-config.rst [hunk 1]
```diff
@@ -472,6 +472,52 @@ Applies the transforms defined in the given projection names.
...
+    * - Property
+      - Type
+      - Description
+    * - synthesizeNames
+      - ``boolean``
+      - Whether enums without names should have names synthesized if possible.
+
+This transformer will attempt to convert strings with :ref:`enum traits <enum-trait>`
+into enum shapes. To successfully convert to an enum shape, each enum
+definitions within the source enum trait must have a `name` property, or the
```
Comment (mtdowling) [inline docs/source-2.0/guides/building-models/build-config.rst:495]: Need to use double backticks here

**Review by mtdowling — CHANGES_REQUESTED** [review]

File: docs/source-2.0/guides/building-models/build-config.rst [hunk 2]
```diff
@@ -472,6 +472,52 @@ Applies the transforms defined in the given projection names.
...
+      - ``boolean``
+      - Whether enums without names should have names synthesized if possible.
+
+This transformer will attempt to convert strings with :ref:`enum traits <enum-trait>`
+into enum shapes. To successfully convert to an enum shape, each enum
+definitions within the source enum trait must have a ``name`` property, or the
+`synthesizeNames` transform config option must be enabled. By default, `synthesizeNames`
+is disabled. Additionally, if an enum definition from the enum trait is marked
+as deprecated, the :ref:`deprecated trait <deprecated-trait>` will be applied
+to the resulting enum shape member. If the enum definition is tagged as
```
Comment (gosar) [inline docs/source-2.0/guides/building-models/build-config.rst:499]: Before talking about the special handling for "internal" tag, mention that all tags from enum definition becomes `@tags` trait on the member.

**Review by gosar — APPROVED** [review]

File: docs/source-2.0/guides/building-models/build-config.rst [hunk 3]
```diff
@@ -472,6 +472,52 @@ Applies the transforms defined in the given projection names.
...
+      - Type
+      - Description
+    * - synthesizeNames
+      - ``boolean``
+      - Whether enums without names should have names synthesized if possible.
+
+This transformer will attempt to convert strings with :ref:`enum traits <enum-trait>`
+into enum shapes. To successfully convert to an enum shape, each enum
+definitions within the source enum trait must have a ``name`` property, or the
+`synthesizeNames` transform config option must be enabled. By default, `synthesizeNames`
```
Comment (mtdowling) [inline docs/source-2.0/guides/building-models/build-config.rst:496]:
```suggestion
``synthesizeNames`` transform config option must be enabled. By default, ``synthesizeNames``
```

**Comment by gosar** [comment]
This transform is available to 1.0 IDL too, since we load everything as 2.0 and serialize as 2.0 anyways. Should we document this in the 1.0 docs too?

**Pull request merged**